sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```
@@ -467,10 +494,170 @@ The `node` class provides the {crs}[common reference semantics].
+more than one dynamic parameter instance.
+
+Registration is done by using the argument index (known when argument
+originally set using `set_arg()`/`set_args()`).
+
+See <<executable-graph-update, Executable Graph Update>> for more information
+about updating node parameters.
+
+The `dynamic_parameter` class provides the {crs}[common reference semantics].
+
+Table {counter: tableNumber}. Member functions of the `dynamic_parameter` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
+[source,c++]
+----
+dynamic_parameter(command_graph<graph_state::modifiable> graph);
+----
+|Constructs a dynamic parameter object that can be registered with node
+arguments.
+
+Parameters:
+
+* `graph` - Graph containing the nodes which will have dynamic parameters.
+
+|
+[source,c++]
+----
+dynamic_parameter(const ValueT &initialValue, command_graph<graph_state::modifiable> graph);
+----
+|Constructs a dynamic parameter object that can be registered with node
+arguments, while providing an intiial value which is used solely for
+deducing ValueT.
+
+Parameters:
+
+* `initialValue` - Initial value used for CTAD.
+* `graph` - Graph containing the nodes which will have dynamic parameters.
+
+|
+[source,c++]
+----
+void register_with_node(const handler &cgh, int argIndex);
+----
+|Callable only from command-group scope when adding a node to a graph
+using the explicit API. Associate the dynamic parameter
+with the kernel parameter of the node that encapsulates this command-group.
+Parameters are associated by their index which was given when they were
+registered with `handler`.
+
+Parameters:
+
+* `handler` - SYCL handler object.
+* `argIndex` - Kernel argument index starting from zero to associate with
+  the dynamic parameter.
+
+Exceptions:
+
+* Throws with error code `invalid` if `argIndex` is not a valid argument index
+  for the resulting node.
+* Throws with error code `invalid` if the type of the command-group is not a
+  kernel execution.
```

This is not implementable I think, since most likely this function is called before the `parallel_for` etc. are called on the `handler`, the `handler` is not aware of what type of `node` it will produce.

We can throw this error during `handler::finalize()` so perhaps this error can be moved to the `graph::add()` definition.